apps/audits/api.py

```diff
@@ -74,6 +75,21 @@ def get_queryset(self):
         queryset = queryset.exclude(type=Types.upload_file).filter(instant=False)
         return queryset
 
+    def perform_update(self, serializer):
+        job = self.get_object()
+        is_periodic = serializer.validated_data.get('is_periodic')
+        if job.is_periodic != is_periodic:
+            job.is_periodic = is_periodic
+            job.save()
+        name, task, args, kwargs = job.get_register_task()
+        task_obj = PeriodicTask.objects.filter(name=name).first()
+        if task_obj:
+            is_periodic = job.is_periodic
+            if task_obj.enabled != is_periodic:
+                task_obj.enabled = is_periodic
+                task_obj.save()
+        return super().perform_update(serializer)
+
 
 class FTPLogViewSet(OrgModelViewSet):
     model = FTPLog
```

Comment:
I don't see any obvious irregularities in this code snippet. However, it seems Django's new-style ORM isn't being used properly since `Q`s from django.db.models.Q aren't used in the `.exclude()` method call on line number 2. This can lead to database queries that might not be optimized or correct.

If you're using a newer version of Django than I'm familiar with, please ensure that both models (FTPLog and periodicTasks) have been created and modified correctly with the right permissions for viewing/uploading files, logging activities, etc.

To optimize the usage of django-celery-beat, make sure celerybeat tasks are enabled on appends 'interval' field when defined and the corresponding jobs table entries have the appropriate statuses based on their scheduled execution intervals.

There are no clear issues; however, checking each piece of data against relevant specifications could uncover more precise problems such as missing columns, incorrect field formats, etc. The suggested solution would depend entirely upon context but likely involves making changes in models.py.